reframe/utility/__init__.py

```diff
@@ -310,3 +311,47 @@ def __eq__(self, *args, **kwargs):
 
     def __ne__(self, *args, **kwargs):
         return self.__mapping.__ne__(*args, **kwargs)
+
+
+class Colorizer(abc.ABC):
+    """Base class for colorizing strings on the terminal."""
+
+    def __repr__(self):
+        return debug.repr(self)
+
+    @abc.abstractmethod
+    def colorize(string, foreground, background):
```

Comment:
Should this be also `@staticmethod`?

Reply:
You mean `@classmethod`? Yes, you have a point.

Reply:
No I mean the `@staticmethod` since you don't need access to the class itself.

Reply:
Actually yes, the `@classmethod` would be even better to access the corresponding escape sequences, etc.

Reply:
I'll take a look and fix it.

Reply:
I refactored completely the coloring code.